integration-tests-jvm/mapstruct/src/main/java/org/apache/camel/quarkus/component/mapstruct/it/MapStructResource.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.mapstruct.it;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.quarkus.component.mapstruct.it.model.Car;
+import org.apache.camel.quarkus.component.mapstruct.it.model.Vehicle;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Path("/mapstruct")
+@ApplicationScoped
+public class MapStructResource {
+    private static final Vehicle VEHICLE = new Vehicle("Volvo", "XC60", "true", 2021);
+
+    @Inject
+    ProducerTemplate producerTemplate;
+
+    @Path("/component")
+    @GET
+    public void componentTest() {
+        testMapping("component");
+    }
+
+    @Path("/converter")
+    @GET
+    public void converterTest() {
+        testMapping("converter");
+    }
+
+    private void testMapping(String endpoint) {
+        Car car = producerTemplate.requestBody("direct:" + endpoint, VEHICLE, Car.class);
+        assertEquals(car.getBrand(), VEHICLE.getCompany());
+        assertEquals(car.getModel(), VEHICLE.getName());
+        assertEquals(car.getYear(), VEHICLE.getYear());
+        assertEquals(car.isElectric(), Boolean.parseBoolean(VEHICLE.getPower()));
```

Comment:
As noted above it would be better to have both these assertions and VEHICLE = new Vehicle("Volvo", "XC60", "true", 2021); in the test class. To avoid adding quarkus-resteasy-jackson or quarkus-resteasy-jsonb (because we really do not want any additional dependencies), you can use some add-hoc serialization, like you pass "Volvo,XC60,true,2021" over HTTP and create the Car out of the String.split(",") array.